README.md

```diff
@@ -3,336 +3,93 @@
+
+## Security Information
+- Audit info TODO
+- Bug bounty info TODO
+- Security contact info TODO
+
+## Protocol Specification
+- [Introduction](#introduction-to-the-protocol)
+- [Trust Model](#trust-model)
+- [Documentation](TODO)
+- [Cucumber Features](./test/features)
+- Glossary _(coming soon)_
+
+### Introduction to the Protocol
+The core of Valorem is the Settlement Engine, which follows the [ERC-1155 multi-token](https://eips.ethereum.org/EIPS/eip-1155) standard. Options can be written for any pair of valid ERC-20 assets (excluding rebasing, fee-on-transfer, and ERC-777 tokens). When written, an options contract is represented by semi-fungible Option tokens, which can be bought/sold/transferred between addresses like any ERC-1155 token.
+
+The option writer's claim to the underlying asset(s) (if not exercised) and exercise asset(s) (if exercised) is represented by a non-fungible Option Lot Claim token. This Claim NFT can be redeemed for their share of the underlying plus exercise assets, based on current. The Settlement Engine uses a internal day bucketing approach, to facilitate fair assignment while minimizing O(n) complexity and gas costs.
+
+The structure of an options contract is as follows:
+
+- **Underlying asset:** the ERC-20 address of the asset to be received
+- **Underlying amount:** the amount of the underlying asset contained within an option contract of this type
+- **Exercise asset:** the ERC-20 address of the asset needed for exercise
+- **Exercise amount:** the amount of the exercise asset required to exercise this option
+- **Earliest exercise timestamp:** the timestamp after which this option may be exercised
+- **Expiry timestamp:** the timestamp before which this option must be exercised
+
+The Valorem core protocol is unopinionated on the type of option (call vs. put), where, when, or for how much an option is bought/sold, and whether or not the option is profitable when exercised. Because all contracts are fully collateralized, options written on the Settlement Engine are TODO.
+
+Read the [Valorem litepaper](https://valorem.xyz/docs/valorem-options-litepaper/) for more insight into the business use cases that Valorem enables.
+
+### Trust Model
+#### Actors
+There are 3 main actors in the core protocol:
+- Protocol Admin
+- Option Writer
+- Option Holder
+
+The **Protocol Admin** is the address to which protocol fees are swept, and can update the Protocol Admin address. No other permissioned actions are possible.
+
+**Option Writers** can create new option types and write option lots for any valid ERC-20 asset pair (excluding rebasing, fee-on-transfer, and ERC-777 tokens). Sufficient approval must be granted for the Settlement Engine to take custody of the requisite amount of the underlying asset. Once an option lot expires, the writer can redeem their Claim NFT to their share of the underlying and/or exercise assets.
```

Comment:
Are we actually excluding these tokens?

Reply:
Not to my knowledge. How do we want to handle? cc @0xAlcibiades

Reply:
Hmm. Let's think about how to phrase, right now it isn't clear.

Reply:
Any ideas here?

Reply:
Not sure the best way to call this out. It's a security assumption. Maybe a chat/voice on Discord for few minutes would be helpful. We maybe could consider permissionless / semi-permissionless ideas to prevent these types of tokens?